WcaOnRails/app/models/championship.rb

```diff
@@ -15,16 +16,46 @@ class Championship < ApplicationRecord
                                 inclusion: { in: CHAMPIONSHIP_TYPES }
 
   def name
-    return "World Championship" if championship_type == "world"
+    return I18n.t('competitions.competition_form.championship_types.world') if championship_type == "world"
 
-    return "Greater China Championship" if championship_type == "greater_china"
+    return I18n.t('competitions.competition_form.championship_types.greater_china') if championship_type == "greater_china"
 
     continent = Continent.c_find(championship_type)
-    return "Continental Championship for #{continent.name}" if continent
+    return I18n.t('competitions.competition_form.championship_types.continental', continent: continent.name) if continent
 
     country = Country.find_by_iso2(championship_type)
-    return "National Championship for #{country.name}" if country
+    return I18n.t('competitions.competition_form.championship_types.national', country: country.name) if country
 
-    "Championship for #{championship_type}"
+    I18n.t('competitions.competition_form.championship_types.generic', type: championship_type)
+  end
+
+  def national?
+    Country.find_by_iso2(championship_type)
```

Comment:
nit: These helper methods have some duplicated code from `def name` up above. It *might* be nice to refactor things so `name` can use these as well. I'm thinking something like: `def country`, which returns the country if it's a national championship, and nil if it's not.

Feel free to agree or disagree with me here about if this makes the code better.

Reply:
Yeah, this did cross my mind,  but that leaves me with the question of how to handle world and Greater China. For word I suppose I could do something like `def planetary` but that seems a bit goofy given that I don't think we anticipate any other planetary championships. I really have no idea what to do for Greater China. Or, would you suggest just leaving those two as booleans and just having country and continent?

Reply:
I think for Greater China, the `greater_china?` method you've already written fits in fine here. Just change the existing line of code to:

```ruby
return I18n.t('competitions.competition_form.championship_types.greater_china') if greater_china?
```

If we ever add a second type of cross-country championship type, we can revisit this and maybe do something like:

```ruby
return I18n.t("competitions.competition_form.championship_types.#{championship_type}") if cross_country_championship?
```

but this might require a bit of i18n work to avoid confusing our scripts that detect if all keys are defined and used, so I don't think it's worth doing right now.

Reply:
For worlds, I would suggest doing something similar, just rewrite the code to do:

```ruby
return I18n.t('competitions.competition_form.championship_types.world') if world?
```

Reply:
Ok, so to clarify, you are suggesting that I leave the `world?` and `greater_china?` boolean methods and also add the string-returning methods here? And same for country/continent? Or were you just using `... if world?` as shorthand and that would be replaced by the code checking to see if it's a world championship?

Reply:
Oh, sorry, I was suggesting real code that could replace the existing code in the `name` method up above. No changes to the `world?` and `greater_china?` method you've already written.